src/store/reducers/network.js

```diff
@@ -169,58 +177,45 @@ const syncNetworkEpic = () => async (dispatch, getStore) => {
  *
  */
 
-const CheckNetwork = createPromiseActions(CHECK_NETWORK);
+const CheckNetworkAction = createPromiseActions(CHECK_NETWORK);
 
-/**
- * @dev Here we create 3 actions for checking the network status
- * @type {{pending, fulfilled, rejected}|*}
- */
-const subscribeLatestBlockFilter = createPromiseActions(
-  'NETWORK/SUBSCRIBE_LATEST_BLOCK_FILTER',
-);
 
 /**
  * @dev We get latest mined block number
  */
 const getLatestBlockNumber = createAction(
   'NETWORK_GET_LATEST_BLOCK_NUMBER',
-  async () => new Promise((resolve, reject) =>
-    web3.eth.getBlockNumber((e, latestBlockNumber) => {
-      if (e) {
-        reject(e);
-      }
-      else {
-        resolve(latestBlockNumber);
-      }
-    }),
-  ),
+  async () => window.web3p.eth.getBlockNumber(),
 );
 
 const getLatestBlock = createAction(
   'NETWORK_GET_LATEST_BLOCK',
-  async () => new Promise((resolve, reject) =>
-    web3.eth.getBlock('latest', (e, res) => {
-      if (e) {
-        reject(e);
-      }
-      else {
-        resolve(res);
-      }
-    }),
-  ),
+  async () => window.web3p.eth.getBlock('latest'),
+);
+
+
+/**
+ * @dev Here we create 3 actions for checking the network status
+ * @type {{pending, fulfilled, rejected}|*}
+ */
+const subscribeLatestBlockFilterAction = createPromiseActions(
+  'NETWORK/SUBSCRIBE_LATEST_BLOCK_FILTER',
 );
+const subscribeLatestBlockFilterEpic = () => async (dispatch) => {
+  dispatch(subscribeLatestBlockFilterAction.pending());
 
-const subscribeLatestBlockFilterEpic = () => (dispatch) => {
-  dispatch(subscribeLatestBlockFilter.pending());
-  window.web3p.eth.filter('latest').then(
-    latestBlockHash => dispatch(getLatestBlockNumber(latestBlockHash)),
-    rej => dispatch(subscribeLatestBlockFilter.rejected(rej))
-  );
-  dispatch(subscribeLatestBlockFilter.fulfilled());
+  window.web3.eth.filter('latest', (e) => {
```

Comment:
Why do we have `window.web3p` and `window.web3`? What is the difference?

Reply:
the web3p is patched with promises

Reply:
Why don't we use one of those only ?

Reply:
Cause its not always possible to use promises.